app/templates/src/main/java/package/social/config/_SocialConfiguration.java

```diff
@@ -0,0 +1,109 @@
+package <%=packageName%>.social.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+import org.springframework.security.crypto.encrypt.Encryptors;
+import org.springframework.social.UserIdSource;
+import org.springframework.social.config.annotation.ConnectionFactoryConfigurer;
+import org.springframework.social.config.annotation.EnableSocial;
+import org.springframework.social.config.annotation.SocialConfigurer;
+import org.springframework.social.connect.ConnectionFactoryLocator;
+import org.springframework.social.connect.UsersConnectionRepository;
+import org.springframework.social.connect.jdbc.JdbcUsersConnectionRepository;
+import org.springframework.social.connect.web.ProviderSignInController;
+import org.springframework.social.connect.web.ProviderSignInUtils;
+import org.springframework.social.connect.web.SignInAdapter;
+import org.springframework.social.facebook.connect.FacebookConnectionFactory;
+import org.springframework.social.google.connect.GoogleConnectionFactory;
+import org.springframework.social.security.AuthenticationNameUserIdSource;
+import org.springframework.social.twitter.connect.TwitterConnectionFactory;
+
+import javax.inject.Inject;
+import javax.sql.DataSource;
+
+/**
+ * Basic Spring Social configuration.  Creates the beans necessary to manage Connections to social services and
+ * link accounts from those services to internal Users.
+ */
+@Configuration
+@EnableSocial
+public class SocialConfiguration implements SocialConfigurer {
+    private final Logger log = LoggerFactory.getLogger(SocialConfiguration.class);
+
+    @Inject
+    private DataSource dataSource;
+
+    @Override
+    public void addConnectionFactories(ConnectionFactoryConfigurer connectionFactoryConfigurer, Environment environment) {
+        // Google configuration
+        String googleClientId = environment.getProperty("spring.social.google.clientId");
```

Comment:
To have type safety, can you use a Spring property bean for this? Like the JHipsterProperties -> I guess there is one provided out of the box by Spring Boot for this